crates/ika-core/src/dwallet_mpc/crytographic_computation/mpc_computations.rs

```diff
@@ -410,10 +449,19 @@ impl ProtocolCryptographicData {
                         malicious_parties,
                         private_output,
                     } => {
-                        // Wrap the public output with its version.
-                        let public_output_value = bcs::to_bytes(
-                            &VersionedDwalletDKGSecondRoundPublicOutput::V1(public_output_value),
-                        )?;
+                        let decentralized_output: <AsyncProtocol as twopc_mpc::dkg::Protocol>::DecentralizedPartyDKGOutput = bcs::from_bytes(&public_output_value)?;
+                        let decentralized_output = match decentralized_output {
+                            DKGDecentralizedPartyVersionedOutputSecp256k1::UniversalPublicDKGOutput {
+                                output, ..
+                            } => output,
+                            DKGDecentralizedPartyVersionedOutputSecp256k1::TargetedPublicDKGOutput (
+                                output
+                            ) => output,
+                        };
+                        let public_output_value =
+                            bcs::to_bytes(&VersionedDwalletDKGSecondRoundPublicOutput::V1(
+                                bcs::to_bytes(&decentralized_output).unwrap(),
+                            ))?;
```

Comment:
This should only be in the `v1` patchy case, add a todo here to check protocol config before